src/platforms/common/px4_log.c

```diff
@@ -106,22 +106,39 @@ __EXPORT void px4_log_modulename(int level, const char *moduleName, const char *
 
 			if (is_atty) { pos += snprintf(buffer + pos, max_length - pos, "%s", __px4_log_level_color[level]); }
 
+			if (pos >= max_length) { return; }
```

Comment:
But if this check happens, we have already overflowed the buffer, right?

Reply:
Exactly.

Reply:
And nothing bad happens then?

Reply:
No, because we pass the right buffer size to `snprintf`. I ran into the case where it was overflowing, which is how I found the problem and this fixed it then.
The only thing that is not nice here is the return. We could split into several messages, but the code would get really messy.